### PR TITLE
[ Client ] Send to a client a different websocket server url if provided 

### PR DESCRIPTION
**Pull request by evaliyev**
At the moment, clients connect to Alfred to create documents using the REST interface and then connect again with WebSockets to send and receive deltas.
This PR added an option to provide a URL different from the orderer to connect to the delta stream WebSocket server URL.

WebSocket server can be specified in the config.json file. If not provided, the WebSocket server URL defaults to Alfred URL.

### CODE REVIEW

**Comment by evaliyev** [comment]
This PR is dependent on https://github.com/microsoft/FluidFramework/pull/11428

**Comment by evaliyev** [comment]
@znewton PTAL

File: packages/drivers/routerlicious-driver/src/documentService.ts [hunk 1]
```diff
@@ -49,6 +49,7 @@ export class DocumentService implements api.IDocumentService {
         private _resolvedUrl: api.IResolvedUrl,
         protected ordererUrl: string,
         private deltaStorageUrl: string,
+        private readonly deltaStreamUrl: string,
```
Comment (znewton) [inline packages/drivers/routerlicious-driver/src/documentService.ts:52]: This needs to be updated in refreshDiscovery so it shouldn't be readonly

File: packages/drivers/routerlicious-driver/src/documentServiceFactory.ts [hunk 2]
```diff
@@ -223,6 +223,7 @@ export class RouterliciousDocumentServiceFactory implements IDocumentServiceFact
             fluidResolvedUrl,
             ordererUrl,
             deltaStorageUrl,
+            fluidResolvedUrl.endpoints.deltaStreamUrl,
```
Comment (znewton) [inline packages/drivers/routerlicious-driver/src/documentServiceFactory.ts:226]: should default to ordererUrl for backcompat

**Comment by znewton** [comment]
Looks like you need to build this locally and push up the API MD changes and also probably update type validations

**Review by jatgarg — APPROVED** [review]

**Review by znewton — APPROVED** [review]

File: server/tinylicious/src/routes/ordering/documents.ts [hunk 3]
```diff
@@ -49,6 +49,7 @@ export function create(storage: IDocumentStorage): Router {
             defaultHash,
             `http://${request.hostname}`,
             `http://${request.hostname}`,
+            `http://${request.hostname}`,
```
Comment (znewton) [inline server/tinylicious/src/routes/ordering/documents.ts:52]: Perhaps upgrade T9s server dependencies and make this change in a separate PR

File: packages/drivers/local-driver/src/localDocumentServiceFactory.ts [hunk 4]
```diff
@@ -76,7 +76,7 @@ export class LocalDocumentServiceFactory implements IDocumentServiceFactory {
             defaultHash,
             resolvedUrl.endpoints.ordererUrl ?? "",
             resolvedUrl.endpoints.storageUrl ?? "",
-            resolvedUrl.endpoints.deltaStreamUrl ?? resolvedUrl.endpoints.ordererUrl ?? "",
+            resolvedUrl.endpoints.deltaStreamUrl ?? (resolvedUrl.endpoints.ordererUrl ?? ""),
```
Comment (znewton) [inline packages/drivers/local-driver/src/localDocumentServiceFactory.ts:79]: nit: you can remove these parenthesis

**Review by znewton — APPROVED** [review]
